rfc/0006-threat-model.md

```diff
@@ -0,0 +1,262 @@
+
+**To do**
+We may came up with some tests to cover this rules reviews
+We  should carefully review a PR changing the rules in the policies deployment
+
+## Threat 11 - Attacker deploys workloads to namespaces that are  exempt from admission control
+An attacker is able to deploy workloads to Kubernetes namespaces that are exempt
+from the  admission controller configuration.
+
+**Mitigation**
+RBAC rights are strictly controlled
+
+**To do**
+I think most of the RBAC is not responsability. But we can help our users if we:
+- Warning them in our docs and *suggest* some minimum RBAC to be used.
+- Provide a policy which detect RBAC changes and **maybe** block them. Is this possible?
+
+
+## Threat ID 12 - Block rule can be bypassed due to missing match (e.g.  missing initcontainers)
+An attacker created a workload manifest which uses a feature of the Kubernetes
+API which  is not covered by the admission controller
+
+**Mitigation**
+All rules are reviewed and tested.
+
+**To do**
+We may came up with some tests to cover this rules reviews
+We  should carefully review a PR changing the rules in the policies deployment
+
+## Threat ID 13 - Attacker exploits bad string matching on a blocklist to  bypass rules
+An attacker, who has rights to create workloads, bypasses a rule by exploiting
+bad string  matching.
+
+**Mitigation**
+All rules are reviewed and tested.
+
+**To do**
+We may came up with some tests to cover this rules reviews
+We  should carefully review a PR changing the rules in the policies deployment
+
+## Threat ID 14 - Attacker uses new/old features of the Kubernetes API  which have no rules
+An attacker, with rights to create workloads, uses new features of the Kubernetes
+API (for  example a changed API version) to bypass a rule.
+
+**Mitigation**
+All rules are reviewed and tested.
+
+**To do**
+We may came up with some tests to cover this rules reviews and API versions.
+We can create a configuration to reject by default requests where the API
+version not cover by the policy.  We should warning policies developers to cover
+all the supported API version in theirs tests and reject all of others.
+
+
+## Threat ID 15 - Attacker deploys privileged container to node running  Webhook controller
+An attacker, who has rights to deploy privileged containers to the cluster, creates
+a privileged  container on the cluster node where the admission controller webhook operates.
+
+**Mitigation**
+Admission controller uses restrictive policies to prevent privileged  workloads
+
+**To do**
+We have a policy to prevent privileged workloads in the recommended policies in
+the `kubewarden-default` helm chart. But we can improve by:
+- We may need to define all the API versions cover by the recommended policies
+- We can create a configuration to reject by default requests where the API version not cover by the policy.
+- We should warning policies developers to cover all the supported API version in theirs tests and reject all of others.
```

Comment:
These don't apply here. Leftover?

Reply:
We provide policies that can help with this and that can be deployed in the default `policy-server` namespace, where all policy server deployments are created.

Reply:
I agree with what rafa wrote

Reply:
Yes, copy and past issue. :man_facepalming:

Reply:
Furthermore, our policies cannot help on this. The threat said container deployed in the cluster node. Kubewarden does not have access to this information. Image a bad actors with access to the node running the controller, we cannot block them from running a privileged container in the node.